src/settings/StaffSlips/StaffSlipDetail.test.js

```diff
@@ -95,71 +85,118 @@ describe('StaffSlipDetail', () => {
+
+    it('should render "Template content" Accordion', () => {
+      expect(Accordion).toHaveBeenNthCalledWith(2, expect.objectContaining({
+        id: mocktemplateContentId,
+        open: accordionDefaultStatus.templateContent,
+      }), {});
+    });
+
+    it('should call Metadata component', () => {
+      expect(Metadata).toHaveBeenCalledWith(expect.objectContaining({
+        connect: testStripes.connect,
+        metadata: testDefaultProps.initialValues.metadata,
+      }), {});
+    });
 
-        fireEvent.click(screen.getByTestId(testIds.staffSlipsPreviewButton));
+    it('should display StaffSlipAboutSection', () => {
+      expect(screen.getByText('StaffSlipAboutSection')).toBeVisible();
+    });
 
-        expect(PreviewModal).toHaveBeenLastCalledWith(expect.objectContaining({
-          open: true,
+    describe('handleExpandAll method', () => {
+      it('should render components with default accordions statuses', () => {
+        expect(ExpandAllButton).toHaveBeenLastCalledWith(expect.objectContaining({
+          accordionStatus: accordionDefaultStatus,
         }), {});
 
-        fireEvent.click(screen.getByTestId(testIds.previewModal));
+        expect(Accordion).toHaveBeenLastCalledWith(
+          expect.objectContaining({
+            open: accordionDefaultStatus.generalInformation,
+          }), {}
+        );
+      });
+
+      it('should expand all accordions statuses', () => {
+        fireEvent.click(screen.getByTestId(mockTestIds.expandAllButton));
 
-        expect(PreviewModal).toHaveBeenLastCalledWith(expect.objectContaining({
-          open: false,
+        expect(ExpandAllButton).toHaveBeenLastCalledWith(expect.objectContaining({
+          accordionStatus: {
+            generalInformation: false,
+            templateContent: false,
+          },
         }), {});
+
+        expect(Accordion).toHaveBeenLastCalledWith(
+          expect.objectContaining({
+            open: false,
+          }), {}
+        );
       });
     });
   });
+
+  describe('handleSectionToggle method', () => {
+    it('should close accordion', () => {
+      Accordion.mockImplementationOnce(({ onToggle, children }) => (
+        // eslint-disable-next-line jsx-a11y/click-events-have-key-events, jsx-a11y/no-static-element-interactions
+        <div
+          data-testid={mockTestIds.accordion}
+          onClick={() => onToggle({ id: mockGeneralStaffSlipDetailId })}
+        >
+          {children}
+        </div>
+      ));
+
+      render(
+        <StaffSlipDetail {...testDefaultProps} />
+      );
+
+      expect(Accordion).toHaveBeenCalledWith(
+        expect.objectContaining({
+          open: accordionDefaultStatus.generalInformation,
+        }), {}
+      );
+
+      fireEvent.click(screen.getByTestId(mockTestIds.accordion));
+
+      expect(Accordion).toHaveBeenCalledWith(
+        expect.objectContaining({
+          open: !accordionDefaultStatus.generalInformation,
+        }), {}
+      );
+    });
+  });
```

Comment:
Could you split it into 2 tests?

Reply:
@artem-blazhko
here Accordion is mocked with "mockImplementationOnce", hence I think it makes sense to have tests this way.

Reply:
@Terala-Priyanka we discussed this case. Here you can create 2 tests, first will check that Accordion is rendered and closed or opened. The second test can contain "mockImplementationOnce" and "click" event.

Reply:
@artem-blazhko 
Tests to verifiy that Accordion has been rendered are include in tests before this test suite - "handleSectionToggle method".
This set of tests ensure that the rendering of Accordion and the click event based on the Accordion mock that have been defined in this specific test. 
Your suggestion above does not intend to test the rending of Accordion using the mockImplementation.

Reply:
@Terala-Priyanka disagrees with you. 
But if you want to leave it as is, let it be.